### PR TITLE
Add alerting for VPA

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/17860

This PR:

- adds basic deployment not satisfied alerts for WC VPA
- removes the `-unique` suffix from a few places where it was still being used

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/daemonset.management-cluster.rules.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ spec:
     - alert: ManagementClusterDaemonSetNotSatisfiedLudacris
       annotations:
         description: '{{`Daemonset {{ $labels.namespace}}/{{ $labels.daemonset }} is not satisfied.`}}'
-      expr: kube_daemonset_status_number_unavailable{cluster_type="management_cluster", cluster_id!="giraffe|argali", daemonset=~"calico-node|kube-proxy|cert-exporter|net-exporter|node-exporter-app-unique"} > 0
+      expr: kube_daemonset_status_number_unavailable{cluster_type="management_cluster", cluster_id!="giraffe|argali", daemonset=~"calico-node|kube-proxy|cert-exporter|net-exporter|node-exporter-app-.+"} > 0
```
Comment (stone-z) [inline helm/prometheus-rules/templates/alerting-rules/daemonset.management-cluster.rules.yml:40]: `node-exporter` has the version number in the daemonset now, like:
```
$ kubectl get ds -A | grep node-ex
giantswarm    node-exporter-app-v1-0-1   7         7         7       7            7           <none>                   6d10h
```

File: helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml [hunk 2]
```diff
@@ -68,7 +68,7 @@ spec:
       annotations:
         description: '{{`Deployment {{ $labels.namespace}}/{{ $labels.deployment }} is not satisfied.`}}'
         opsrecipe: deployment-not-satisfied/
-      expr: kube_deployment_status_replicas_unavailable{cluster_type="management_cluster", deployment!~"aws-operator-.+|azure-operator-.+|cluster-operator-.+|coredns-.+|nginx-ingress-controller-.+|worker-.+|master-.+|app-operator-.+|chart-operator-.+|.+-admission-controller-unique|alertmanager.*|prometheus.*|promxy.*", cluster_id!~"argali|giraffe"} > 0
+      expr: kube_deployment_status_replicas_unavailable{cluster_type="management_cluster", deployment!~"aws-operator-.+|azure-operator-.+|cluster-operator-.+|coredns-.+|nginx-ingress-controller-.+|worker-.+|master-.+|app-operator-.+|chart-operator-.+|.+-admission-controller|alertmanager.*|prometheus.*|promxy.*", cluster_id!~"argali|giraffe"} > 0
```
Comment (stone-z) [inline helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml:71]: I've checked on `gauss` that there aren't any deployments using the `-unique` suffix, including admission controllers specifically. Just to be on the safe side, pinging @giantswarm/team-biscuit for your thumbs up

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ spec:
       annotations:
         description: '{{`Prometheus can''t communicate with Kubernetes API.`}}'
         opsrecipe: prometheus-cant-communicate/
-      expr: rate(prometheus_sd_kubernetes_http_request_total{app!="promxy-app-unique", status_code="<error>"}[15m]) > 0.25
+      expr: rate(prometheus_sd_kubernetes_http_request_total{app!="promxy-app", status_code="<error>"}[15m]) > 0.25
```
Comment (stone-z) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:17]: Also dropping the `unique` suffix from `promxy` after checking on `gauss` that the promxy app no longer has the suffix. Pinging @giantswarm/team-atlas so you're aware

**Comment by stone-z** [comment]
Tested on `gauss` by making VPA's life miserable in a workload cluster :fire:

File: helm/prometheus-rules/templates/alerting-rules/deployment.workload-cluster.rules.yml [hunk 4]
```diff
@@ -26,7 +26,7 @@ spec:
       annotations:
         description: '{{`Deployment {{ $labels.namespace}}/{{ $labels.deployment }} is not satisfied.`}}'
         opsrecipe: workload-cluster-deployment-not-satisfied/
-      expr: kube_deployment_status_replicas_unavailable{cluster_type="workload_cluster", deployment="metrics-server"} > 0
+      expr: kube_deployment_status_replicas_unavailable{cluster_type="workload_cluster", deployment=~"metrics-server|vertical-pod-autoscaler-app-admission-controller|vertical-pod-autoscaler-app-recommender|vertical-pod-autoscaler-app-updater"} > 0
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/deployment.workload-cluster.rules.yml:29]: Why would Kaas teams be alerted when VPA does not run on a workload cluster ? Are we using it now as part of a release ?

Also, would it make sense to page when it's not working on Management clusters as well ?

Comment (MarcelMue) [reply helm/prometheus-rules/templates/alerting-rules/deployment.workload-cluster.rules.yml:29]: Okay I am confused:

Ludacris got this issue about owning VPA https://github.com/giantswarm/giantswarm/issues/17860

But according to the ownership doc it belongs to atlas:
https://intranet.giantswarm.io/docs/product/team-ownership/

What went wrong here? cc @puja108

**Comment by stone-z** [comment]
So we discussed with Puja in Ludacris and while we don't have a problem taking VPA alerts, we think it makes sense for them to go to Atlas. Originally, VPA alerts were going to be Ludacris because some customers rely on it in WCs, but it isn't actually a core feature, it's an optional App. Ordinarily, we'd then suggest it go to Balo, but since Atlas are the internal users/owners of VPA, we thought it might be problematic if Ludacris or Balo were to make a change for WCs which then breaks VPA for the Atlas usage of it.

So, we suggest they go to Atlas. Open to discuss in SIG Product if needed

**Comment by JosephSalisbury** [comment]
eh? i thought we'd gone w/ ludacris owning vpa in https://github.com/giantswarm/giantswarm/pull/17858?

**Comment by QuentinBisson** [comment]
We could take ownership of VPA in the MCs but WCs are out of scope to me

**Comment by stone-z** [comment]
@JosephSalisbury @puja108 plz discuss in SIG Product and let me know :laughing:

**Comment by stone-z** [comment]
Alright, took them back to Ludacris for the time being. @giantswarm/team-ludacris can I get an approval?

**Review by MarcelMue — APPROVED** [review]

**Comment by kopiczko** [comment]
Feels like VPA in MC should be owned by biscuit TBH.

**Comment by MarcelMue** [comment]
> Feels like VPA in MC should be owned by biscuit TBH.

We decided to go for this in Planning on monday. There are still product discussions going on - I believe also on thursday. We just didn't want to get blocked here any longer.

**Comment by kopiczko** [comment]
Yeah I got that. Just added my 2c.

**Pull request merged**